### PR TITLE
Fix Email test issue discovered in #5271.

### PR DESCRIPTION
**Pull request by bummzack**
Updated/added tests for changed- and forgot-password Emails.
Updated fixture and tests to no longer use a real Email address.

### CODE REVIEW

**Comment by bummzack** [comment]
@dhensby As discussed in #5271, this patches the issue with Email testing for 3. Will create a PR for 3.2 as well if you agree to these changes.

**Comment by dhensby** [comment]
@bummzack if you create a PR against 3.2 then it'll be merged into 3 and master - you don't need individual PRs for each branch :)

NB: I know it's already in master

File: tests/security/MemberTest.php [hunk 1]
```diff
@@ -185,16 +185,44 @@ public function testPasswordChangeLogging() {
 	 * Test that changed passwords will send an email
 	 */
 	public function testChangedPasswordEmaling() {
+		$oldSetting = Config::inst()->get('Member', 'notify_password_change');
+		Config::inst()->update('Member', 'notify_password_change', true);
```
Comment (dhensby) [inline tests/security/MemberTest.php]: You don't need to store the old settings, the most you have to do is nest and unnest the config like so:

```php
function testThing() {
    Config::nest();
    //test code
    Config::unnest();
}
```

But I think we actually made each test "sandboxed" now so the nesting and unnesting is done automatically.

File: tests/security/MemberTest.yml [hunk 2]
```diff
@@ -42,7 +42,7 @@ Member:
   test:
     FirstName: Test
     Surname: User
-    Email: sam@silverstripe.com
+    Email: dummy@silverstripe.tld
```
Comment (dhensby) [inline tests/security/MemberTest.yml]: I'd prefer using `@example.com` which will never become a real domain rather than one that just isn't real yet.

Comment (bummzack) [inline tests/security/MemberTest.yml]: Oh, ok. I thought .tld was "protected" from becoming a real TLD as well, but I guess that's not true and you never know with ICANN. These changes here are in line with the ones I made to the master branch though… is this something that should be changed in master as well?

**Comment by bummzack** [comment]
Closed in favor of #5298 (which is against 3.2)

File: tests/security/MemberTest.yml (hunk 2 above)
Comment (dhensby) [inline tests/security/MemberTest.yml]: We can just merge them in when we do the merge from 3.2 -> 3 -> master

**Pull request closed without merging**